flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -72,20 +76,32 @@ public void write(RowData row) throws IOException {
 
     switch (row.getRowKind()) {
       case INSERT:
+        if (upsert) {
+          LOG.error("+I UPSERT");
+        }
+        LOG.error("+I - INSERT row: {}", row);
+        writer.write(row);
+        break;
+      // Need to check here when the equalityFieldIds are not matching the table spec.
       case UPDATE_AFTER:
+        LOG.error("+U - UPDATE_AFTER row: {}", row);
         if (upsert) {
           writer.delete(row);
+          break;
         }
         writer.write(row);
         break;
 
       case UPDATE_BEFORE:
+        LOG.error("-U - UPDATE_BEFORE row: {}", row);
         if (upsert) {
+          writer.delete(row);
           break;  // UPDATE_BEFORE is not necessary for UPDATE, we do nothing to prevent delete one row twice
```

Comment:
It _seemed_ like this was the appropriate place to add the `delete` call, but on `upsert` every record is coming in as `+I`. You can see it in the logs.

Ignore reading data for now. Right now we're focused more on fixing data for the future. cc @hililiwei